--- includes/class-rum.php
+++ includes/class-rum.php
@@ -192,7 +192,7 @@
				return;
			}

-			$parsed_path = isset( $_SERVER['REQUEST_URI'] ) ? wp_parse_url( wp_unslash( $_SERVER['REQUEST_URI'] ), PHP_URL_PATH ) : '/';
+			$parsed_path = isset( $_SERVER['REQUEST_URI'] ) ? wp_parse_url( sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) ), PHP_URL_PATH ) : '/';
			$path        = $parsed_path ? esc_url_raw( $parsed_path ) : '/';

			$config = array(
